createBookFromReadme.sh

```diff
@@ -18,15 +18,33 @@ function cleanupBeforeStarting(){
 
 # Splits the Readme.md file based on the header in markdown and creates chapters
 # Note:
-#   Get gcsplit via homebrew on mac: brew install coreutils
+#   Get csplit via homebrew on mac: brew install coreutils
 function splitIntoChapters(){
+    OS_NAME=$(uname | tr '[:upper:]' '[:lower:]')
+
+    case $OS_NAME in
+  linux*)
+    OS_NAME='linux'
+    csplit --prefix='Chapter_' --suffix-format='%d.md' --elide-empty-files README.md '/^## /' '{*}' -q
+    ;;
+  darwin*)
+    OS_NAME='osx'
     gcsplit --prefix='Chapter_' --suffix-format='%d.md' --elide-empty-files README.md '/^## /' '{*}' -q
+    ;;
+  freebsd*)
+    OS_NAME='freebsd'
+    csplit --prefix='Chapter_' --suffix-format='%d.md' --elide-empty-files README.md '/^## /' '{*}' -q
+    ;;
+  *)
+    OS_NAME=notset
+    ;;
+esac
```

Comment:
I personally donot prefer the switch statements in bash scripts. They are hard to read compared to simpler if-elif statements.

You could have done something like below and executed the specific command:

```bash
if [[ "$OSTYPE" == "linux-gnu" ]]; then
    # ...
elif [[ "$OSTYPE" == "darwin"* ]]; then
    # Mac OSX
elif [[ "$OSTYPE" == "cygwin" ]]; then
    # POSIX compatibility layer and Linux environment emulation for Windows
elif [[ "$OSTYPE" == "msys" ]]; then
    # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
elif [[ "$OSTYPE" == "freebsd"* ]]; then
    #FreeBSD
else
    # Unknown.
fi
```